lib/src/firebase_auth_mocks_base.dart

```diff
@@ -38,10 +39,21 @@ class MockFirebaseAuth extends Mock implements FirebaseAuth {
     return _fakeSignIn();
   }
 
-  @override Future<AuthResult> signInWithCustomToken({@required String token}) {
+  @override
+  Future<AuthResult> signInWithCustomToken({@required String token}) {
     return _fakeSignIn();
   }
 
+  @override
+  Future<AuthResult> signInAnonymously() {
+    return _fakeSignIn();
+  }
+
+  @override
+  Future<void> signOut() async {
+    _currentUser = null;
```

Comment:
It'd be nice to call `stateChangedStreamController.add(null);` as well so that whoever listens to `onAuthStateChanged` gets notified. I've filed #11 to keep track of it.